backend/src/main.rs

```diff
@@ -332,6 +341,12 @@ async fn get_stats(
     }))
 }
 
+async fn get_default_game_settings(
+    setting: game_state::PropagatedState,
+) -> Result<impl warp::Reply, warp::Rejection> {
+    Ok(warp::reply::json(&setting))
```

Comment:
I think you can just have

```
Ok(warp::reply::json(&game_state::PropagatedState::default()))
```

and not take `setting` as an argument -- might make it simpler?

Reply:
oh, I tried that, but don't know how to skip .and().

Reply:
```diff --git a/backend/src/main.rs b/backend/src/main.rs
index 6225a12..3c17a26 100644
--- a/backend/src/main.rs
+++ b/backend/src/main.rs
@@ -243,6 +243,7 @@ async fn main() {
     let game_stats = warp::path("stats")
         .and(games)
         .and_then(|(game, stats)| get_stats(game, stats));
+    let default_settings = warp::path("default_settings.json").and_then(default_propagated);
     let routes = index
         .or(js)
         .or(js_map)
@@ -252,11 +253,16 @@ async fn main() {
         .or(api)
         .or(rules)
         .or(dump_state)
-        .or(game_stats);
+        .or(game_stats)
+        .or(default_settings);

     warp::serve(routes).run(([127, 0, 0, 1], 3030)).await;
 }

+async fn default_propagated() -> Result<impl warp::Reply, warp::Rejection> {
+    Ok(warp::reply::json(&game_state::PropagatedState::default()))
+}
+
 async fn dump_state(games: Games) -> Result<impl warp::Reply, warp::Rejection> {
     let mut state_dump: HashMap<String, game_state::GameState> = HashMap::new();
     let mut games = games.lock().await;```

should work

Reply:
Thanks. You must have good hands! I did the same, and vscode complained! It is probably a delayed validation from vscode